### PR TITLE
feat: Implement generation of app-level FastAPI module for multiple notebooks

### PR DESCRIPTION
**Pull request by asymness**
Resolves https://github.com/Unstructured-IO/community/issues/18

**Summary of changes**:
- [x] Implement generation of an app-level "central" module that imports and runs notebook-level modules.
- [x] Update unit tests and ensure 100% coverage.
- [x] Update version and changelog.

**Test Instructions:**
Install `unstructured-api-tools==0.4.4`.

1. In `pipeline-sec-filings` project, add a new notebook called `pipeline-test.ipynb` and add a cell with the following contents:
    ```python
    # pipeline-api
    
    def pipeline_api(text, file=None, m_var=[]):
        return {"message": "Got request", "var": m_var}
    ```
2. Run `make generate-api`.

- Run the following command to run the standalone `test` notebook API:
    ```
    PYTHONPATH=. uvicorn prepline_sec_filings.api.test:app --reload
    ```

- Run the following command to run the standalone `section` notebook API:
    ```
    PYTHONPATH=. uvicorn prepline_sec_filings.api.section:app --reload
    ```

- Run the following command to run the central `app` notebook API that runs both `section` and `test` notebook APIs:
    ```
    PYTHONPATH=. uvicorn prepline_sec_filings.api.app:app --reload
    ```

### CODE REVIEW

File: unstructured_api_tools/pipelines/convert.py [hunk 1]
```diff
@@ -56,6 +58,11 @@ def _infer_params_from_pipeline_api(script: str) -> Dict[str, Optional[Any]]:
     """A helper function to prepare jinja interpolation.
     Returns a list of string (multi-value) parameters to expose in the FastAPI route.
     """
+    if script.count("def pipeline_api(") > 1:
```
Comment (yuming-long) [inline unstructured_api_tools/pipelines/convert.py:61]: could you add tests for these two cases? you can test the logger info with `caplog` from `pytest`.

File: unstructured_api_tools/pipelines/convert.py [hunk 2]
```diff
@@ -56,6 +58,11 @@ def _infer_params_from_pipeline_api(script: str) -> Dict[str, Optional[Any]]:
...
+        logging.warn("Function pipeline_api was redefined in the pipeline API definition.")
+    elif script.count("def pipeline_api(") < 1:
```
Comment (yuming-long) [inline unstructured_api_tools/pipelines/convert.py:63]: same as comment above :)

**Review by yuming-long — CHANGES_REQUESTED** [review]
Looks really good! and tested with curl commands that all three APIs could work.

File: unstructured_api_tools/pipelines/convert.py (hunk 1 above)
Comment (asymness) [reply unstructured_api_tools/pipelines/convert.py:61]: Sure, I have added. Thanks for the suggestion!

**Comment by yuming-long** [comment]
LGTM!

**Review by yuming-long — APPROVED** [review]

**Pull request merged**